### ISSUE TITLE
wrong-number-of-arguments (3 . 4) 2 when trying to load slime-helper.el in 29.0.50

### ISSUE THREAD

**Issue by stuart-little** [quicklisp/quicklisp-slime-helper#27]
I have 
- `sbcl 2.2.2`
- `slime 2.26.1`
- `emacs 29.0.50` compiled from [source](https://git.savannah.gnu.org/git/emacs.git/)

Evaluating 
```
(load (expand-file-name "~/quicklisp/slime-helper.el"))
```
results in the following error:
```
Debugger entered--Lisp error: (wrong-number-of-arguments (3 . 4) 2)
  #f(compiled-function (obsolete-name current-name when &optional docstring) "Make OBSOLETE-NAME a variable alias for CURRENT-NAME and mark it obsolete.\n\nWHEN should be a string indicating when the variable was first\nmade obsolete, for example a date or a release number.\n\nThis macro evaluates all its parameters, and both OBSOLETE-NAME\nand CURRENT-NAME should be symbols, so a typical usage would look like:\n\n  (define-obsolete-variable-alias 'foo-thing 'bar-thing \"28.1\")\n\nThis macro uses `defvaralias' and `make-obsolete-variable' (which see).\nSee the Info node `(elisp)Variable Aliases' for more details.\n\nIf CURRENT-NAME is a defcustom or a defvar (more generally, any variable\nwhere OBSOLETE-NAME may be set, e.g. in an init file, before the\nalias is defined), then the define-obsolete-variable-alias\nstatement should be evaluated before the defcustom, if user\ncustomizations are to be respected.  The simplest way to achieve\nthis is to place the alias statement before the defcustom (this\nis not necessary for aliases that are autoloaded, or in files\ndumped with Emacs).  This is so that any user customizations are\napplied before the defcustom tries to initialize the\nvariable (this is due to the way `defvaralias' works).\n\nFor the benefit of Customize, if OBSOLETE-NAME has\nany of the following properties, they are copied to\nCURRENT-NAME, if it does not already have them:\n`saved-value', `saved-variable-comment'." #<bytecode 0x7de324727241818>)('common-lisp-glossary-fun 'common-lisp-hyperspec-glossary-function)
  (define-obsolete-variable-alias 'common-lisp-glossary-fun 'common-lisp-hyperspec-glossary-function)
  eval-buffer(#<buffer  *load*-885161> nil "$HOME/quicklisp/dists/quicklisp/softwa..." nil t)  ; Reading at buffer position 45354
  load-with-code-conversion("$HOME/quicklisp/dists/quicklisp/softwa..." "$HOME/quicklisp/dists/quicklisp/softwa..." nil t)
  require(hyperspec "lib/hyperspec")
  eval-buffer(#<buffer  *load*-93494> nil "$HOME/quicklisp/dists/quicklisp/softwa..." nil t)  ; Reading at buffer position 2866
  load-with-code-conversion("$HOME/quicklisp/dists/quicklisp/softwa..." "$HOME/quicklisp/dists/quicklisp/softwa..." nil t)
  (slime-setup '(slime-fancy))
  (let* ((quicklisp-slime-directory (quicklisp-slime-helper-slime-directory))) (add-to-list 'load-path quicklisp-slime-directory) (require 'slime-autoloads) (setq slime-backend (expand-file-name "swank-loader.lisp" quicklisp-slime-directory)) (setq slime-path quicklisp-slime-directory) (slime-setup '(slime-fancy)))
  eval-buffer(#<buffer  *load*> nil "$HOME/quicklisp/slime-helper.el" nil t)  ; Reading at buffer position 1547
  load-with-code-conversion("$HOME/quicklisp/slime-helper.el" "$HOME/quicklisp/slime-helper.el" nil nil)
  load("$HOME/quicklisp/slime-helper.el")
  (progn (load (expand-file-name "~/quicklisp/slime-helper.el")))
  eval((progn (load (expand-file-name "~/quicklisp/slime-helper.el"))) t)
  elisp--eval-last-sexp(nil)
  eval-last-sexp(nil)
  funcall-interactively(eval-last-sexp nil)
  call-interactively(eval-last-sexp nil nil)
  command-execute(eval-last-sexp)
```

**Comment by jrciii** [comment]
@stuart-little I solved this issue on emacs 28.1 by running this in sbcl:

```
 (ql:update-dist "quicklisp")
```

This upgraded slime and then emacs started fine.

**Comment by stuart-little** [comment]
@jrciii Thanks! Yes, that worked just now (still on `Emacs 29.0.50`).
